src/target/adiv5.c

```diff
@@ -554,7 +560,10 @@ void adiv5_dp_init(ADIv5_DP_t *dp)
 				stm32_prepare(ap);
 			}
 		}
-		adiv5_component_probe(ap, ap->base, false, 0, 0);
```

Comment:
Here, finally adiv5_component_probe fails because no entry in the ROM table is present, but  ap->pidr is set and we can finally check if we have an Atmel DSU, and call the probing function (maybe a call more specific is needed, like kinetis_mdm_probe or nrf51_mdm_probe)